src/vumi2/routers.py

```diff
@@ -60,23 +67,36 @@ async def setup(self):
 
     # TODO: Teardown
 
+    async def _get_matched_mapping_names(self, addr):
+        matched_names = []
+        for name, pattern in self.mappings:
+            if pattern.match(addr):
+                matched_names.append(name)
+
+        if matched_names == [] and self.config.default_app:
+            matched_names.append(self.config.default_app)
```

Comment:
The append implies that we might add the default app even if the matched list is non-empty. Would something like
```python
if self.config.default_app and not matched_names:
    return [self.config.default_app]
```
perhaps be clearer?